test/specs/test-util.el

```diff
@@ -46,6 +46,36 @@
         (goto-char (point-min))
         (narrow-to-region (point-min) (point-min))
         (expect (buffer-string) :to-equal "")
-        (expect (flycheck-buffer-empty-p) :not :to-be-truthy)))))
+        (expect (flycheck-buffer-empty-p) :not :to-be-truthy))))
+
+  (describe "flycheck-buffer-saved-p"
+
+    (it "considers a buffer not corresponding to a file unsaved"
+      (with-temp-buffer
+        (expect (flycheck-buffer-saved-p) :not :to-be-truthy)))
+
+    (it "considers a modified buffer not corresponding to a file unsaved"
+      (with-temp-buffer
+        (insert "modification")
+        (expect (flycheck-buffer-saved-p) :not :to-be-truthy)))
+
+    (it "considers an unmodified buffer corresponding to a file saved"
+      (spy-on 'file-exists-p :and-call-fake (lambda (&optional buffer) t))
+      (spy-on 'buffer-file-name :and-call-fake (lambda (&optional buffer) "my-buffer-name"))
+      (with-temp-buffer
+        (expect (flycheck-buffer-saved-p) :to-be-truthy))
+      (expect (spy-calls-count 'buffer-file-name) :to-equal 1)
+      (expect (spy-calls-count 'buffer-file-name) :to-equal 1)
```

Comment:
This condition is duplicated?  I guess you wanted to check `file-exists-p` on the second, right?